camel/configs.py

```diff
@@ -11,12 +11,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from abc import ABC
 from dataclasses import dataclass, field
-from typing import Dict, Optional, Sequence, Union
+from typing import Any, Dict, List, Optional, Sequence, Union
 
 
 @dataclass(frozen=True)
-class ChatGPTConfig:
+class BaseConfig(ABC):
```

Comment:
Good! See, you kept the type check on place and made dependency inversion to loosen the coupling of agents and model configs. The final step would be to move BaseConfig to a file of its own, but we can skip it for now.